src/Compilers/Core/Portable/SourceFileResolver.cs

```diff
@@ -66,7 +66,7 @@ public SourceFileResolver(ImmutableArray<string> searchPaths, string baseDirecto
                         throw new ArgumentException(CodeAnalysisResources.NullValueInPathMap, nameof(pathMap));
                     }
 
-                    if (IsPathSeparator(key[key.Length - 1]))
+                    if (PathUtilities.IsDirectorySeparator(key[key.Length - 1]))
```

Comment:
>PathUtilities.IsDirectorySeparator [](start = 24, length = 34)

Just realized -- this changes semantics on Unix. The helper in PathUtilities uses the current platform separator(s), but here we always want both / and \